### PR TITLE
Refactor Spark33XShims to avoid code duplication

### LINKED ISSUE #5150: [BUG] Optimize Spark33XShims to reduce code deplicate
**Describe the bug**

Some operators of ANSI interval types were copied from `GpuOverrides` only with the differentce of type checks.
It would be nice to shim the types to pass rather than make a copy of the entire override that has to be kept in sync other than just which types to pass as parameters. This applies elsewhere where we're making a copy only to update the type checks.

Check all the operators in `Spark33XShims`

**Additional context**
Refer to https://github.com/NVIDIA/spark-rapids/pull/5020
This PR shimed the types for some operators like UnaryMinus, Add, Subtract, UnaryPositive ......, need to check others.

### PR DESCRIPTION
**Pull request by res-life**
Closes #5150 

Refector Spark33XShims

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
A known issue caused Check Markdown links failed
[✖] https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark-tools_2.12/22.04.0/ → Status: 404

File: sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/Spark320PlusShims.scala [hunk 1]
```diff
@@ -337,93 +337,14 @@ trait Spark320PlusShims extends SparkShims with RebaseShims with Logging {
...
-          override def tagPlanForGpu(): Unit = {
-            if (!p.runtimeFilters.isEmpty) {
-              willNotWorkOnGpu("runtime filtering (DPP) on datasource V2 is not supported")
-            }
-          }
-
-          override def convertToGpu(): GpuExec =
-            GpuBatchScanExec(p.output, childScans.head.convertToGpu(), p.runtimeFilters)
-        })
+        (p, conf, parent, r) => new BatchScanExecMeta320Plus(p, conf, parent, r))
```
Comment (firestarman) [inline sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/Spark320PlusShims.scala:347]: NIT:
Personally I prefer `BatchScanExecMeta` to `BatchScanExecMeta320Plus`.

**Comment by firestarman** [comment]
LGTM except a nit as the above.

**Comment by res-life** [comment]
build

File: sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/Spark320PlusShims.scala (hunk 1 above)
Comment (res-life) [reply sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/Spark320PlusShims.scala]: Done

**Review by jlowe — APPROVED** [review]

**Pull request merged**